### ISSUE TITLE
Add support for "doas".

### ISSUE THREAD

**Issue by somewhatfrog** [Dreaming-Codes/nvidia_oc#4]
Currently it doesn't work if "sudo" is not present in the system and have to be run as "doas".
Though it is not a big deal if run as service it seems, otherwise app works great!
Thanks!

**Comment by Dreaming-Codes** [comment]
This https://github.com/Dreaming-Codes/nvidia_oc/commit/8d4374bc57099c4912890f2f3a18662d1168ff8b should have added doas support but I've not tested it. Let me know if it works.
That commit first appears in version 0.1.12

**Comment by somewhatfrog** [comment]
```
thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sudo2-0.2.1/src/lib.rs:174:52:
failed to execute child: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Dreaming-Codes** [comment]
Well then it means that sudo2 only supports doas on openBSD. You should open an issue on its repo to https://github.com/dilawar/sudo2

**Comment by Dreaming-Codes** [comment]
I think I know what I've missed for doas support I'll take at it later when I'm back at my hotel

**Comment by somewhatfrog** [comment]
Sorry, completely forgot about this, yes it seems to work well now, closing! Thank you, this thing is great!

**Comment by somewhatfrog** [comment]
sorry my bad, forgot to run it with parameters, issue is still present

**Comment by Dreaming-Codes** [comment]
> sorry my bad, forgot to run it with parameters and thought it is fixed, issue is still present

I completely forgot too, I'll push a fix soon

**Comment by Dreaming-Codes** [comment]
@somewhatfrog could you please try it now?

**Comment by somewhatfrog** [comment]
`nvidia_oc set --index 0 --freq-offset 195 --mem-offset 1600
thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sudo2-0.2.1/src/lib.rs:174:52:
failed to execute child: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

**Comment by Dreaming-Codes** [comment]
@somewhatfrog oh, it appears that sudo2 panic instead of returning an error... I should have addressed this in 0.1.14 let me know if it works

**Comment by somewhatfrog** [comment]
works as of 0.1.15, closing

thank you!
